src/DIRAC/Resources/Computing/ARCComputingElement.py

```diff
@@ -208,8 +208,13 @@ def _addCEConfigDefaults(self):
         ComputingElement._addCEConfigDefaults(self)
 
     #############################################################################
-    def __writeXRSL(self, executableFile):
-        """Create the JDL for submission"""
+    def __writeXRSL(self, executableFile, inputs=None, outputs=None):
```

Comment:
In https://github.com/DIRACGrid/DIRAC/pull/5542 there's the new AREX CE which is just a different ARC configuration of an ARC CE. I am scared that changes here would not be reflected to AREX CE.

Reply:
Why not? I don't get it.
I have already interacted with AREX services using the ARC CE and the `emies` interface.
I didn't try via the REST interface but I don't see why it would not be possible.

Reply:
Because it is coded as:
```
class AREXComputingElement(ComputingElement):
```
and not as
```
class AREXComputingElement(ARCComputingElement):
```

Maybe something that @rajanandakumar should consider.

Reply:
There are two ways of going about this ...

1. Port the changes over to the AREX computing element. Is it only the _writeXRSL function that has changed? Currently such changes mean that the changes will need to be made in both CEs as the code is duplicated.
2. Move to inheriting the ARCComputingElement within the AREXComputingElement.

It depends on what you plan for the future of the ARCComputing element.

Reply:
Option 2 looks like the least error-prone to me.

Reply:
This seems to need work. When I try option 2, I get the following error.

```
2021-11-02 18:41:47 UTC WorkloadManagement/SiteDirector-uk DEBUG: Trying to load DIRAC.Resources.Computing.AREXComputingElement
2021-11-02 18:41:47 UTC WorkloadManagement/SiteDirector-uk ERROR: Agent exception while calling method beginExecution
Traceback (most recent call last):
  File "/opt/dirac/versions/v10.3.0a5-1634708729/Linux-x86_64/lib/python3.9/site-packages/DIRAC/Core/Base/AgentModule.py", line 342, in am_secureCall
    result = functor(*args)
  File "/opt/dirac/versions/v10.3.0a5-1634708729/Linux-x86_64/lib/python3.9/site-packages/LHCbDIRAC/WorkloadManagementSystem/Agent/SiteDirector.py", line 31, in beginExecution
    res = DIRACSiteDirector.beginExecution(self)
  File "/opt/dirac/versions/v10.3.0a5-1634708729/Linux-x86_64/lib/python3.9/site-packages/DIRAC/WorkloadManagementSystem/Agent/SiteDirector.py", line 242, in beginExecution
    result = getQueuesResolved(
  File "/opt/dirac/versions/v10.3.0a5-1634708729/Linux-x86_64/lib/python3.9/site-packages/DIRAC/WorkloadManagementSystem/Utilities/QueueUtilities.py", line 74, in getQueuesResolved
    result = ceFactory.getCE(ceName=ce, ceType=ceDict["CEType"], ceParametersDict=ceQueueDict)
  File "/opt/dirac/versions/v10.3.0a5-1634708729/Linux-x86_64/lib/python3.9/site-packages/DIRAC/Resources/Computing/ComputingElementFactory.py", line 48, in getCE
    result = ObjectLoader().loadObject("Resources.Computing.%s" % subClassName)
  File "/opt/dirac/versions/v10.3.0a5-1634708729/Linux-x86_64/lib/python3.9/site-packages/DIRAC/Core/Utilities/ObjectLoader.py", line 106, in loadObject
    result = self.loadModule(importString, hideExceptions=hideExceptions)
  File "/opt/dirac/versions/v10.3.0a5-1634708729/Linux-x86_64/lib/python3.9/site-packages/DIRAC/Core/Utilities/ObjectLoader.py", line 94, in loadModule
    result = self.__rootImport(importString, hideExceptions=hideExceptions)
  File "/opt/dirac/versions/v10.3.0a5-1634708729/Linux-x86_64/lib/python3.9/site-packages/DIRAC/Core/Utilities/ObjectLoader.py", line 74, in __rootImport
    result = recurseImport(impName, hideExceptions=hideExceptions)
  File "/opt/dirac/versions/v10.3.0a5-1634708729/Linux-x86_64/lib/python3.9/site-packages/DIRAC/Core/Utilities/Extensions.py", line 194, in recurseImport
    return S_OK(importlib.import_module(modName))
  File "/opt/dirac/versions/v10.3.0a5-1634708729/Linux-x86_64/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/dirac/versions/v10.3.0a5-1634708729/Linux-x86_64/lib/python3.9/site-packages/DIRAC/Resources/Computing/AREXComputingElement.py", line 45, in <module>
    class AREXComputingElement(ARCComputingElement):
TypeError: module() takes at most 2 arguments (3 given)
```
Any ideas?

Reply:
Okay - tested and it works in the AREXComputingElement. I will commit the code in a short while.

However, I note that there are a few cleanups / improvements that probably should happen in the ARCComputingElement - including in the function __writeXRSL - based on the feedback I got for the AREXComputingElement which is based on the ARCComputingElement.